server/src/main/java/org/elasticsearch/index/analysis/IndexAnalyzers.java

```diff
@@ -54,9 +55,9 @@ public IndexAnalyzers(IndexSettings indexSettings, NamedAnalyzer defaultIndexAna
         this.defaultIndexAnalyzer = defaultIndexAnalyzer;
         this.defaultSearchAnalyzer = defaultSearchAnalyzer;
         this.defaultSearchQuoteAnalyzer = defaultSearchQuoteAnalyzer;
-        this.analyzers = unmodifiableMap(analyzers);
-        this.normalizers = unmodifiableMap(normalizers);
-        this.whitespaceNormalizers = unmodifiableMap(whitespaceNormalizers);
+        this.analyzers =  unmodifiableMap(new HashMap<>(analyzers));
+        this.normalizers = unmodifiableMap(new HashMap<>(normalizers));
+        this.whitespaceNormalizers = unmodifiableMap(new HashMap<>(whitespaceNormalizers));
```

Comment:
I added this to make sure whatever class is passing in the analyzers/normalizers/whitespaceNormalizers map cannot hold on to a rerence to the map and modify it after creation. As far as I can see we currently don't do this (e.g. in AnalysisRegistry#build the input maps are all discarded after use), but this is a defensive copy to prevent any accidental references to the underlying maps leaking. I think this makes reasoning about the behaviour of this class much sager and shouldn't add too much overhead.